src/System.IO.Packaging/src/System/IO/Packaging/PackUriHelper.cs

```diff
@@ -379,15 +481,165 @@ internal static string GetStringForPartUri(Uri partUri)
             return validatedUri.PartUriString;
         }
 
+        //This method validates the packUri and returns its two components if they are valid-
+        //1. Package Uri
+        //2. Part Uri
+        internal static void ValidateAndGetPackUriComponents(Uri packUri, out Uri packageUri, out Uri partUri)
+        {
+            //Validate if its not null and is an absolute Uri, has pack:// Scheme.
+            packUri = ValidatePackUri(packUri);
+            packageUri = GetPackageUriComponent(packUri);
+            partUri = GetPartUriComponent(packUri);
+        }
+
         #endregion Internal Methods
-        
+
+        #region Private Constructor
+
+        static PackUriHelper()
+        {
+            EnsurePackSchemeRegistered();
+        }
+
+        #endregion Private Constructor
+
         #region Private Methods
 
         private static readonly char[] HexUpperChars = {
             '0', '1', '2', '3', '4', '5', '6', '7',
             '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'
         };
 
+        /// <summary>
+        /// This method is used to create a valid pack Uri
+        /// </summary>
+        /// <param name="packageUri">This is the uri that points to the entire package.
+        /// This parameter should be an absolute Uri. This parameter cannot be null or empty </param>
+        /// <param name="partUri">This is the uri that points to the part within the package
+        /// This parameter should be a relative Uri.
+        /// This parameter can be null in which case we will create a valid pack uri
+        /// that references the entire package</param>
+        /// <param name="fragment">Fragment for the resulting Pack URI. This parameter can be null
+        /// The fragment string must start with a "#"</param>
+        /// <returns>A Uri with the "pack://" scheme</returns>
+        /// <exception cref="ArgumentNullException">If packageUri parameter is null</exception>
+        /// <exception cref="ArgumentException">If packageUri parameter is not an absolute Uri</exception>
+        /// <exception cref="ArgumentException">If partUri parameter does not conform to the valid partUri syntax</exception>
+        /// <exception cref="ArgumentException">If fragment parameter is empty or does not start with a "#"</exception>
+        private static Uri Create(Uri packageUri, Uri partUri, string fragment)
+        {
+            // Step 1 - Validate input parameters
+            packageUri = ValidatePackageUri(packageUri);
+
+            if (partUri != null)
+                partUri = ValidatePartUri(partUri);
+
+            if (fragment != null)
+            {
+                if (fragment == String.Empty || fragment[0] != '#')
+                    throw new ArgumentException(SR.FragmentMustStartWithHash);
```

Comment:
Nit: could we add a parameter name to this?

Reply:
the parameter name was baked into the resource string, but i'll bring it out so i can use SR.Format.